### PR TITLE
Server directory move

### PR DESCRIPTION
**Pull request by kuzdogan**
This, again 😬, ended up being a big PR but it's actually straightforward.

But I love it because it removes so much

- Removes the s3 and ipfs services, CI runs, and related content
- removes the .env files from `environments/` and moves each to the respective folder, mostly to server
- Turns root `package.json` into a `sourcify-monorepo` name
- moves server into services/server
- Unifies linting by removing eslint and configs from all packages and only having it in the root
- Adds linting to monitor

### CODE REVIEW

**Review by marcocastignoli — COMMENTED** [review]
It all seems good but I cannot see the differences in server directly because you moved the files. I ran it and it works.

But for me the "Mocha - Server" debug option doesn't work:

```
Error: ENOENT: no such file or directory, open 'test/testcontracts/Storage/Storage.sol'
    at Object.openSync (node:fs:590:3)
    at Object.readFileSync (node:fs:458:35)
    at Suite.<anonymous> (/Users/marcocastignoli/Projects/sourcify/services/server/test/server.js:77:27)
    at Object.create (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/interfaces/common.js:148:19)
    at context.describe.context.context (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/interfaces/bdd.js:42:27)
    at Object.<anonymous> (/Users/marcocastignoli/Projects/sourcify/services/server/test/server.js:57:1)
    at Module._compile (node:internal/modules/cjs/loader:1191:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1245:10)
    at Module.load (node:internal/modules/cjs/loader:1069:32)
    at Function.Module._load (node:internal/modules/cjs/loader:904:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:530:24)
    at async importModuleDynamicallyWrapper (node:internal/vm/module:438:15)
    at async formattedImport (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/nodejs/esm-utils.js:9:14)
    at async Object.exports.requireOrImport (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/nodejs/esm-utils.js:42:28)
    at async Object.exports.loadFilesAsync (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/nodejs/esm-utils.js:100:20)
    at async singleRun (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/cli/run-helpers.js:125:3)
    at async Object.exports.handler (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/cli/run.js:370:5)
```

The problem seems to be here: sourcify/services/server/test/server.js:77:27

```ts
  const sourcePath = path.join(
    "test",
    "testcontracts",
    "Storage",
    "Storage.sol"
  );
  const sourceBuffer = fs.readFileSync(sourcePath);
```

**Comment by kuzdogan** [comment]
> It all seems good but I cannot see the differences in server directly because you moved the files. I ran it and it works.
> 
> But for me the "Mocha - Server" debug option doesn't work:
> 
> ```
> Error: ENOENT: no such file or directory, open 'test/testcontracts/Storage/Storage.sol'
>     at Object.openSync (node:fs:590:3)
>     at Object.readFileSync (node:fs:458:35)
>     at Suite.<anonymous> (/Users/marcocastignoli/Projects/sourcify/services/server/test/server.js:77:27)
>     at Object.create (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/interfaces/common.js:148:19)
>     at context.describe.context.context (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/interfaces/bdd.js:42:27)
>     at Object.<anonymous> (/Users/marcocastignoli/Projects/sourcify/services/server/test/server.js:57:1)
>     at Module._compile (node:internal/modules/cjs/loader:1191:14)
>     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1245:10)
>     at Module.load (node:internal/modules/cjs/loader:1069:32)
>     at Function.Module._load (node:internal/modules/cjs/loader:904:12)
>     at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
>     at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
>     at async Promise.all (index 0)
>     at async ESMLoader.import (node:internal/modules/esm/loader:530:24)
>     at async importModuleDynamicallyWrapper (node:internal/vm/module:438:15)
>     at async formattedImport (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/nodejs/esm-utils.js:9:14)
>     at async Object.exports.requireOrImport (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/nodejs/esm-utils.js:42:28)
>     at async Object.exports.loadFilesAsync (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/nodejs/esm-utils.js:100:20)
>     at async singleRun (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/cli/run-helpers.js:125:3)
>     at async Object.exports.handler (/Users/marcocastignoli/Projects/sourcify/node_modules/mocha/lib/cli/run.js:370:5)
> ```
> 
> The problem seems to be here: sourcify/services/server/test/server.js:77:27
> 
> ```ts
>   const sourcePath = path.join(
>     "test",
>     "testcontracts",
>     "Storage",
>     "Storage.sol"
>   );
>   const sourceBuffer = fs.readFileSync(sourcePath);
> ```

Yes those need to have '__dirname' instead of 'test' Weird I thought I updated all

File: .circleci/continue_config.yml [hunk 1]
```diff
@@ -72,12 +66,6 @@ aliases:
                   - ./*
       # If no build is needed, echo 0 to run as success 
       - run: echo 0
```
[the remark this answers is not in the document]
Comment (kuzdogan) [reply .circleci/continue_config.yml:68]: It wasn't used anywhere

File: .gitignore [hunk 2]
```diff
@@ -23,4 +22,4 @@ logs
 **/too_big.txt
 chain-tests-report/
 coverage
-!src/server/controllers/repository
\ No newline at end of file
+workspace/
```
[the remark this answers is not in the document]
Comment (kuzdogan) [reply .gitignore:25]: It's where we write the built image sha .txt's on the CI. 

https://github.com/ethereum/sourcify/blob/1e68376f751b78ef87f57012fa5218e4f2ad67bc/scripts/build_and_publish_docker_images.sh#L33

File: package.json [hunk 3]
```diff
@@ -1,45 +1,28 @@
...
+  "scripts": {
+    "build:clean": "rm -rf node_modules services/monitor/node_modules services/server/node_modules packages/bytecode-utils/node_modules/ packages/contract-call-decoder/node_modules/ packages/lib-sourcify/node_modules/ && npm install && npx lerna run build",
+    "build:lerna": "lerna run build",
+    "server:start": "cd services/server && node ./dist/server/server.js",
+    "ui:start": "cd ui && npm run start",
+    "monitor:start": "node ./services/monitor/dist/index.js",
+    "update-chains": "node --experimental-fetch scripts/updateChains.mjs && prettier --write services/server/src/chains.json",
+    "lerna-test": "lerna run test --stream --concurrency 1",
+    "lerna-lint": "lerna run lint",
+    "lerna-version": "lerna version --no-push"
```
[the remark this answers is not in the document]
Comment (kuzdogan) [reply package.json:20]: They are in `services/server`. Do they need to be on the root?

File: packages/bytecode-utils/package.json [hunk 4]
```diff
@@ -45,17 +45,11 @@
   },
   "devDependencies": {
     "@ava/typescript": "^1.1.1",
```
[the remark this answers is not in the document]
Comment (kuzdogan) [reply packages/bytecode-utils/package.json:47]: I think I did but let me do a clean install to make sure

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcocastignoli) [reply package.json:20]: it depends how they are called from the ci, I don't know

**Review by marcocastignoli — APPROVED** [review]

**Pull request merged**